### PR TITLE
[SYCL] Add testing for multi_ptr with extended USM address spaces

### PR DESCRIPTION
**Pull request by steffenlarsen**
This commit splits multi_ptr and multi_ptr_legacy into two tests; one compiled and run as before and another defining
`__ENABLE_USM_ADDR_SPACE__` to define the extended USM address spaces.

This requires https://github.com/intel/llvm/pull/7680.

### CODE REVIEW

File: SYCL/Basic/multi_ptr_usm_addr_ext.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// RUN: %clangxx -D__ENABLE_USM_ADDR_SPACE__ -fsycl -fsycl-targets=%sycl_triple -fsycl-dead-args-optimization %s -o %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+//==--------------- multi_ptr.cpp - SYCL multi_ptr test --------------------==//
```
Comment (KseniyaTikhomirova) [inline SYCL/Basic/multi_ptr_usm_addr_ext.cpp]: seems to be not updated.

**Review by KseniyaTikhomirova — APPROVED** [review]

File: SYCL/Basic/multi_ptr_usm_addr_ext.cpp (hunk 1 above)
Comment (steffenlarsen) [reply SYCL/Basic/multi_ptr_usm_addr_ext.cpp]: Good eye! Updated.

**Pull request merged**